### ISSUE TITLE
Merge inspec-aws into core

### ISSUE THREAD

**Issue by arlimus** [inspec/inspec#2233]
https://github.com/chef/inspec-aws
review AWS inspec resources to get them to a state where they can be comfortably merged into core inspec:

* [x] review CIS benchmark and identify resources https://d0.awsstatic.com/whitepapers/compliance/AWS_CIS_Foundations_Benchmark.pdf
* [ ] develop missing resources
* [ ] review MITRE feedback on AWS
* [ ] combine with InSpec 2.0 Platforms RFC: https://github.com/chef/inspec/issues/1661
    (run as a target? maybe a custom target? credentials handling? ...)
* [ ] docs
* [ ] website

**Comment by clintoncwolfe** [comment]
Here's where AWS is maintaining some CIS benchmark tooling.  May be useful for gap analysis.  https://github.com/awslabs/aws-security-benchmark  Was hoping to find a SCAP file, but it looks like they generate it from a YAML file - mainly focused on Config rules

**Comment by clintoncwolfe** [comment]
Some things we've identified as potential detailed work items:
* [x] identify highest value resources based on CIS controls targeted
* [x] Update and validate development and testing instructions and code
* [x] Check integration testing process for concerns about hijacking; consider mocked AWS infra like `localstack`
* [x] Get temporary target / connection functionality in place while waiting on Platforms API
* [ ] VPC resources, attributes, matchers, filters - MVP for CIS benchmark
* [x] Cloudwatch resources, attributes, matchers, filters - MVP for CIS benchmark
* [ ] (in progress) Config resources, attributes, matchers, filters - MVP for CIS benchmark
* [x] SNS resources, attributes, matchers, filters - MVP for CIS benchmark
* [ ] S3 resources, attributes, matchers, filters - MVP for CIS benchmark
* [ ] (in progress) Cloudtrail resources, attributes, matchers, filters - MVP for CIS benchmark
* [ ] Sweep existing issues and PRs for dupes, existing work
* [ ] Determine merge technique to pull in inspec-aws repo
* [x] Accommodate AWS integration testing in Rakefile, excluded from default integration testing

**Comment by aaronlippold** [comment]
Hi,

Not to worry, we have a parser for that :) We should be sending you a nice
profile starting point within a day or two - just fixing an issue or two
with parselet.

Aaron

--------
Aaron Lippold

lippold@gmail.com

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

**Comment by joerg** [comment]
Is there a timeline for this merge? We are moving to AWS right now and I would love to show our internal security what Inspec with those resources can do.

**Comment by clintoncwolfe** [comment]
There's a lot of activity over on https://github.com/chef/inspec-aws . I've updated my checklist [above](https://github.com/chef/inspec/issues/2233#issuecomment-335592261) with current progress.  We'd like to cover CIS AWS Benchmark as much as possible (about 8 recommendations are not automation-friendly) in the next few weeks.  That won't result in a "general purpose" kit (there is nothing about RDS in CIS, for example) but will give a good start.

Are there particular areas of AWS that you are interested in?  Feel free to open separate issues requesting them, and we can start prioritizing.

Also, keep in mind that you can use the existing inspec-aws work by declaring a dependency on it in your profile's inspec.yml:

```
depends:
  - name: aws
    git: https://github.com/chef/inspec-aws.git
```

That will let you use any [inspec-aws resources](https://github.com/chef/inspec-aws/tree/master/docs/resources) that currently exist.
